benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/HubProtocolBenchmark.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.SignalR.Internal;
+using Microsoft.AspNetCore.SignalR.Internal.Encoders;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+
+namespace Microsoft.AspNetCore.SignalR.Microbenchmarks
+{
+    [ParameterizedJobConfig(typeof(CoreConfig))]
+    public class HubProtocolBenchmark
+    {
+        private HubProtocolReaderWriter _hubProtocolReaderWriter;
+        private byte[] _binaryInput;
+        private TestBinder _binder;
+        private HubMessage _hubMessage;
+
+        [Params(Message.Small, Message.Medium, Message.Large)]
```

Comment:
Use descriptive names rather than Small/Medium/Large. Also, there are kinda two axes here: Number of payload values, size of payload values. We should see if we can cross-product them a little (e.g. a test that serializes a single 4K string, and a test that serializes 4K 1 byte strings). I think aiming for 2-3 "expected scenarios" (e.g. 3-4 small values, 8-10 small values, 1 very large value) and then a couple "extreme" scenarios (lots of small values, one extraordinarily large value) would be good. Assuming that doesn't balloon the run time too much :).

Reply:
That would balloon it like crazy, I think it already takes 13 minutes for this benchmark